dds/DCPS/transport/framework/TransportSendStrategy.cpp

```diff
@@ -1917,6 +1917,39 @@ TransportSendStrategy::mb_to_iov(const ACE_Message_Block& msg, iovec* iov)
   return num_blocks;
 }
 
+bool TransportSendStrategy::fragmentation_helper(
+  TransportQueueElement* original_element, TqeVector& elements_to_send)
+{
+  const size_t space = space_available();
```

Comment:
`space_available()` depends on the current state of the Send Strategy (`header_.length_`) so I don't think it can be used here.  Not sure if that would be enough to fix the failing test.